assets/report_template.Rmd

```diff
@@ -0,0 +1,1444 @@
+        scrollY = "300px",
+        paging = FALSE))
+}
+```
+
+<!-- Section on Preprocessing -->
+
+```{r, eval = !isFALSE(params$mqc_plot) || !isFALSE(params$dada_filtntrim_args), results='asis'}
+cat("# Preprocessing\n")
+```
+
+<!-- Subsection on FastQC / MultiQC -->
+
+```{r, eval = !isFALSE(params$mqc_plot), results='asis'}
+cat(paste0("
+## FastQC
+
+[FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) gives general quality metrics about your sequenced reads.
+It provides information about the quality score distribution across your reads, per base sequence content (%A/T/G/C),
+adapter contamination and overrepresented sequences. The sequence quality was checked using FastQC and resulting data was
+aggregated using the FastQC module of [MultiQC](https://multiqc.info/). For more quality controls and per sample quality checks you can check the full
+MultiQC report, which can be found in [multiqc/multiqc_report.html](../multiqc/multiqc_report.html).
+"))
+```
+
+```{r, eval = !isFALSE(params$mqc_plot), out.width='100%', dpi=1200, fig.align='center'}
+knitr::include_graphics(params$mqc_plot)
+```
+
+<!-- Subsection on Cutadapt -->
+
+```{r, eval = !isFALSE(params$ca_sum_path), results='asis'}
+cat(paste0("
+## Primer removal with Cutadapt
+
+[Cutadapt](https://journal.embnet.org/index.php/embnetjournal/article/view/200) is trimming primer sequences from sequencing reads.
+Primer sequences are non-biological sequences that often introduce point mutations that do not reflect sample sequences. This is especially
+true for degenerated PCR primer. If primer trimming were to be omitted, artifactual amplicon sequence variants might be computed by
+the denoising tool or sequences might be lost due to being labelled as PCR chimera.
+"))
+
+# import tsv
+cutadapt_summary <- read.table(file = params$ca_sum_path, header = TRUE, sep = "\t")
+
+passed_col <- as.numeric(substr(
+        cutadapt_summary$cutadapt_passing_filters_percent, 1, 4))
+
+max_disc <- round( 100 - min(passed_col), 1 )
+avg_passed <- round(mean(passed_col),1)
+
+cutadapt_text_unch <- "Primers were trimmed using cutadapt"
+cutadapt_text_ch <- paste0(" and all untrimmed sequences were discarded. ",
+    "Sequences that did not contain primer sequences were considered artifacts. Less than ",
+    max_disc, "% of the sequences were discarded per sample and a mean of ",
+    avg_passed, "% of the sequences per sample passed the filtering.")
+
+if (!params$flag_retain_untrimmed) cutadapt_text <- paste0(
+    cutadapt_text_unch, cutadapt_text_ch
+    ) else cutadapt_text <- paste0(cutadapt_text_unch, ".")
+
+cat(cutadapt_text)
+
+# shorten header by "cutadapt_" to optimize visualisation
+colnames(cutadapt_summary) <- gsub("cutadapt_","",colnames(cutadapt_summary))
+
+datatable(cutadapt_summary, options = list(
+        scrollX = TRUE,
+        scrollY = "300px",
+        paging = FALSE))
+
+cat("\n\nCutadapt results can be found in folder [cutadapt](../cutadapt).")
```

Comment:
Maybe we should add something like Between x (X%) and y (Y%) reads per sample (in average z, Z%) were retained for analysis. in case of discarding non-trimmed sequences (default)?